### PR TITLE
Sector Storage checkCommit() (checks #3)

### PR DESCRIPTION
**Pull request by Alexey-N-Chernyshov**
Signed-off-by: Alexey-N-Chernyshov <alexey.n.chernyshov@gmail.com>

### Description of the Change
Add sector storage checkCommit()

### CODE REVIEW

File: core/sector_storage/checks/checks.cpp [hunk 1]
```diff
@@ -145,6 +154,68 @@ namespace fc::sector_storage::checks {
...
+                api->StateMinerSectorSize(miner_address, tipset_key));
+    OUTCOME_TRY(seal_proof_type, sealProofTypeFromSectorSize(sector_size));
+
+    if (sector_info.comm_r
+        != state_sector_precommit_info.value().info.sealed_cid) {
+      return ChecksError::kBadSealedCid;
+    }
+    OUTCOME_TRY(
+        verified,
+        Proofs::verifySeal(SealVerifyInfo{
```
Comment (Alexey-N-Chernyshov) [inline core/sector_storage/checks/checks.cpp:200]: Reviewer, please, check parameters.

File: core/sector_storage/checks/checks.cpp [hunk 2]
```diff
@@ -145,6 +154,68 @@ namespace fc::sector_storage::checks {
...
+    if (seed != sector_info.seed) {
+      return ChecksError::kBadSeed;
+    }
+
+    OUTCOME_TRY(sector_size,
+                api->StateMinerSectorSize(miner_address, tipset_key));
+    OUTCOME_TRY(seal_proof_type, sealProofTypeFromSectorSize(sector_size));
+
+    if (sector_info.comm_r
+        != state_sector_precommit_info.value().info.sealed_cid) {
```
Comment (artyom-yurin) [inline core/sector_storage/checks/checks.cpp:195]:
```suggestion
        != state_sector_precommit_info->info.sealed_cid) {
```

File: core/sector_storage/checks/checks.cpp [hunk 3]
```diff
@@ -145,6 +154,68 @@ namespace fc::sector_storage::checks {
...
+    }
+
+    OUTCOME_TRY(state_sector_precommit_info,
+                getStateSectorPreCommitInfo(
+                    miner_address, sector_info, tipset_key, api));
+    if (!state_sector_precommit_info.has_value()) {
+      return ChecksError::kPrecommitNotFound;
+    }
+
+    if (state_sector_precommit_info.value().precommit_epoch
```
Comment (artyom-yurin) [inline core/sector_storage/checks/checks.cpp:173]:
```suggestion
    if (state_sector_precommit_info->precommit_epoch
```

File: core/sector_storage/checks/checks.cpp [hunk 4]
```diff
@@ -145,6 +154,68 @@ namespace fc::sector_storage::checks {
...
+    }
+    OUTCOME_TRY(
+        verified,
+        Proofs::verifySeal(SealVerifyInfo{
+            .sector = SectorId{.miner = miner_address.getId(),
+                               .sector = sector_info.sector_number},
+            .info =
+                OnChainSealVerifyInfo{
+                    .sealed_cid =
+                        state_sector_precommit_info.value().info.sealed_cid,
```
Comment (artyom-yurin) [inline core/sector_storage/checks/checks.cpp:206]:
```suggestion
                        state_sector_precommit_info->info.sealed_cid,
```

**Review by artyom-yurin — APPROVED** [review]

File: core/sector_storage/checks/checks.cpp (hunk 1 above)
Comment (artyom-yurin) [reply core/sector_storage/checks/checks.cpp:200]: LGTM

**Review by turuslan — APPROVED** [review]

**Pull request merged**